### PR TITLE
feat: supported upgradable cellDep detection

### PR DESCRIPTION
**Pull request by zhangyouxin**
### Breaking Change

- The `getCellDep` method in `AbstractProvider` has changed to asynchronous method.

### Changed

- `MercuryProvider` can provide latest celldeps on chain.
- `MercuryProvider` can provide configured future celldeps on chain.

### CODE REVIEW

File: packages/base/src/AbstractProvider.ts [hunk 1]
```diff
@@ -4,28 +4,22 @@ import {
...
-  MIN_FEE_RATE: Hexadecimal;
-};
-export type ProviderConfig = LumosConfig & OptionalConfig;
-export type InitOptions<T extends LumosConfig = LumosConfig> = Omit<T, keyof OptionalConfig> & Partial<OptionalConfig>;
+import { Provider, ProviderConfig, InitOptions, ResolvedOutpoint, CellOutPointProvider } from './';
 
 export abstract class AbstractProvider implements Provider {
   private initialized = false;
   private _config: ProviderConfig | undefined;
+  abstract depOutPointProvider: CellOutPointProvider | undefined;
```
Comment (homura) [inline packages/base/src/AbstractProvider.ts]: maybe we dont need a `depOutPointProvider` in AbstractProvider, override `getCellDep` is fine

File: packages/ckit/src/__tests__/secp256k1-sudt.spec.ts [hunk 2]
```diff
@@ -343,8 +343,7 @@ test('test serialize and deserialized', async () => {
   const serialized = builder.serialize(unsigned);
   const deserialized = TransferCkbBuilder.serde.deserialize(serialized);
 
-  const txHash = await provider.sendTransaction(await genesisSigner.seal(deserialized));
-
+  const txHash = await provider.sendTxUntilCommitted(await genesisSigner.seal(deserialized));
```
Comment (homura) [inline packages/ckit/src/__tests__/secp256k1-sudt.spec.ts:346]: why we need to change it to sendUntilCommitted

Comment (zhangyouxin) [reply packages/ckit/src/__tests__/secp256k1-sudt.spec.ts:346]: This line is the last line in the test case here, I think if we don't wait until commited here, the next test case may take some cell that already been used in this test case, thus causing next test case to fail with a time out error.

Comment (homura) [reply packages/ckit/src/__tests__/secp256k1-sudt.spec.ts:346]: i see, make sense, and this is probably the reason why ckit always failed in CI before

File: packages/base/src/AbstractProvider.ts (hunk 1 above)
Comment (zhangyouxin) [reply packages/base/src/AbstractProvider.ts]: Done, good advice, didn't get used to overriding before.

**Comment by homura** [comment]
maybe we can rename `rc_lock_new` to `rc_lock_{commit_id}` to make it easier to understand

**Comment by zhangyouxin** [comment]
> maybe we can rename `rc_lock_new` to `rc_lock_{commit_id}` to make it easier to understand

rename to `rc_lock_ddaeae8` https://github.com/nervosnetwork/ckb-production-scripts/commit/ddaeae8f065a5b600805b46d75c53de9bf290e68

**Pull request merged**